pytorch_lightning/metrics/metric.py

```diff
@@ -0,0 +1,248 @@
+from abc import ABC, abstractmethod
+from typing import Any, Optional
+
+import torch
+import torch.distributed
+
+from pytorch_lightning.metrics.utils import tensor_metric, numpy_metric
+from pytorch_lightning.utilities.apply_to_collection import apply_to_collection
+
+__all__ = ['AbstractMetric', 'TensorMetric', 'NumpyMetric']
+
+
+class AbstractMetric(torch.nn.Module, ABC):
```

Comment:
```suggestion
class Metric(torch.nn.Module, ABC):
```
I think that the Abstract in the name is redundant... maybe `AnyMetrc`